test/workbox-precaching/node/test-PrecacheController.mjs

```diff
@@ -441,6 +441,31 @@ describe(`[workbox-precaching] PrecacheController`, function() {
       expect(cacheWrapper.put.args[1][0].plugins).to.equal(testPlugins);
     });
 
+    it(`it should use the proper 'this' when calling a cacheWillUpdate plugin`, async function() {
+      sandbox.spy(cacheWrapper, 'put');
+
+      const precacheController = new PrecacheController();
+      const cacheList = [
+        '/index.1234.html',
+      ];
+      precacheController.addToCacheList(cacheList);
+
+      class TestPlugin {
+        cacheWillUpdate({response}) {
+          expect(this).to.be.an.instanceof(TestPlugin);
```

Comment:
Can you use make this a spy and then at the bottom of the test use `spy.thisValues` or `spy.calledOn()` to assert the correct context?

As it is now, if you changed the file to not call `cacheWillUpdate` at all, this test would still pass.

Reply:
Oh, I didn't know about `thisValues`.